use-case-django-continuous-integration.md

```diff
@@ -0,0 +1,148 @@
+This guide shows you how to use Semaphore to set up a continuous integration
+(CI) pipeline for a Django web application.
+Before starting, [create a new Semaphore project][create-project].
+
+## Define the pipeline
+
+Our Django CI pipeline will perform the following tasks:
+
+- Lint code style with Pylint
+- Run tests for each application using the built in framework
+
+Mature applications usually have many tests and optimizing their runtime saves
+a lot of valuable time for development. So for demonstration we'll run
+tests for each application in the project in parallel.
+
+We want fast feedback through the test pipeline, so we'll fail fast if
+the linting fails. This prevents running time consuming tests for
+semantically incorrect code.
+
+Consider a smaple Django `blog` project with two applications: `posts`
+and `comments`. For these reasons our pipeline is composed of three
+blocks, one for setup and two for tests:
+
+<pre><code class="language-yaml"># .semaphore/semaphore.yml
+version: v1.0
+name: Example Django Application
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Setup
+    task:
+      jobs:
+        - name: Cache pip downloads
+          commands:
+            - sem-version python 3.7
+            - checkout
+            - cache restore requirements-$(checksum requirements.txt)
```

Comment:
Would pip benefit from restoring a previous version of deps bundle? If so let's add fallback keys (see Rails guide for example) — cache restore does partial matching.